src/settings.cc

```diff
@@ -202,6 +202,16 @@ void Settings::visit(SettingsVisitor& visitor)
 	}
 }
 
+SettingsEntry* Settings::getSettingEntryByName(std::string name)
+{
+	for (std::list<SettingsEntry *>::iterator it = entries.begin();it != entries.end();it++) {
+		if(((*it)->name().compare(name))== 0){
```

Comment:
You can use the `==` operator. Assuming the for loop gets fixed: `if (entry->name() == name) return entry;`